docker_cli/image.py

```diff
@@ -0,0 +1,458 @@
+            A boolean designating whether or not the docker build should use
+            the cache.
+
+        Returns
+        -------
+        Image
+            The created image.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker build command fails.
+        ValueError
+            If both `dockerfile` and `dockerfile_string` are defined.
+        """
+        ...
+
+    @overload
+    @classmethod
+    def build(
+        cls: Type[Self],
+        tag: str,
+        *,
+        dockerfile_string: str,
+        context: Optional[os.PathLike[str]],
+        stdout: Optional[io.TextIOBase],
+        stderr: Optional[io.TextIOBase],
+        network: str,
+        no_cache: bool
+    ) -> Self:
+        """
+        Builds a new image from a string in dockerfile syntax.
+
+        Parameters
+        ----------
+        tag : str
+            A name for the image.
+        dockerfile_string : str
+            A Dockerfile-formatted string.
+        context : os.PathLike or None, optional
+            The build context. Defaults to ".".
+        stdout : io.TextIOBase or None, optional
+            A file-like object that the stderr output of the docker build
+            program will be redirected to to. If None, no redirection will
+            occur. Defaults to None.
+        stderr : io.TextIOBase or None, optional
+            A file-like object that the stderr output of the docker build
+            program will be redirected to. If None, no redirection will occur.
+            It should be noted that docker build primarily outputs to stderr.
+            Defaults to None.
+        network : str, optional
+            The name of the network. Defaults to "host".
+        no_cache : bool, optional
+            A boolean designating whether or not the docker build should use
+            the cache.
+
+        Returns
+        -------
+        Image
+            The created Image.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker build command fails.
+        ValueError
+            If both `dockerfile` and `dockerfile_string` are defined.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        tag,
+        *,
+        dockerfile=None,
+        dockerfile_string=None,
+        context=".",
+        stdout=None,
+        stderr=None,
+        network="host",
+        no_cache=True
+    ):
+        if dockerfile is not None and dockerfile_string is not None:
+            raise ValueError(
+                "Both dockerfile and dockerfile_string passed as arguments."
+            )
+
+        # Build with dockerfile if dockerfile_string is None
+        dockerfile_build = dockerfile_string is None
+
+        context_str = os.fspath(".") if context is None else os.fspath(context)
+        cmd = [
+            "docker",
+            "build",
+            f"--network={network}",
+            context_str,
+            "-t",
+            tag
```

Comment:
Seems a bit cleaner to combine these two into a single list item.

```suggestion
            f"-t={tag}"
```